src/ai/bot.js

```diff
@@ -113,10 +113,20 @@ export class MCTSBot extends Bot {
       objectives = () => ({});
     }
 
+    if (typeof iterations === 'number') {
```

Comment:
If you move these lines below 129 (and also adjust `this.iterations` / `iterations`), then you don't have to change the code there that's setting the default values.

Reply:
Honestly I think that is just my TypeScript kicking in. Doing it this way means that the constructor arg might be either number or func but the property will only ever be a func. Simpler that way. I can change it if you like, though.

Reply:
Oh I see.  OK, let's leave it as is then.